### PR TITLE
WIP Enable relative img support for html parser; Add _backlinks-attribute…

### PR DESCRIPTION
**Pull request by datakurre**
… to allow rquerying for related images and replacing img tags with gatsby images

_backlinks allow supposedly an optimal way to replace img tags with gatsby images.

* [x] update example project
* [ ] acceptance tests

### CODE REVIEW

**Comment by datakurre** [comment]
I’m now able to replace inline images on Plone document richtext fields with gatsby images (and similarly linking attachments should work). I’m happy :)

**Comment by ajayns** [comment]
This is great! Let me check this out!

**Comment by datakurre** [comment]
I’ll update test project today or tomorrow.

File: src/gatsby-node.js [hunk 1]
```diff
@@ -235,14 +262,17 @@ const processNodesUsingSearchTraversal = async (
     })
   );
 
+  // Mutable container for collecting all backlinks
+  backlinks = {};
```
Comment (ajayns) [inline src/gatsby-node.js]: I think this would be better as `let backlinks = {}`

**Comment by datakurre** [comment]
Here’s now a page with Plone rich text turned into 1) gatsby link 2) gatsby image and 3) link to gatsby-published attachment 

https://collective.github.io/gatsby-source-plone/testcases/a-page-with-internal-linking/

**Comment by ajayns** [comment]
I've been going through the codebase, but I didn't exactly get the purpose of backlinks, I understand what it does from the code, but why is it required? I realized I didn't fully understand this while working on the documentation for HTML parsing

**Comment by datakurre** [comment]
@ajayns When rendering HTML from Plone (parsing it into react tree and then rendering in Document or NewsItem component), we want to replace Plone img-tags with gatsby-images and links to files with links to files fetched into gatsby static folder.

Without backlinks that would require iterating every image or file for each replaced img/a-tag to find the correct one. This would be problem on larger site and would also slow down the generated site, because that would need to be done every time a page with HTML is being rendered in browser.

With backlinks, we can have optimized set of related images and files for each component (calculated on compilation/build time):

* in default layout, we find node for matching path and render it with a proper component, AND also pass the component all related images and files  https://github.com/collective/gatsby-source-plone/blob/master/tests/gatsby-starter-default/src/templates/default.js#L59

* in rich text component (called from Document or NewsItem) we use that data to replace links to files and images with optimized gatsby-images: https://github.com/collective/gatsby-source-plone/blob/master/tests/gatsby-starter-default/src/components/RichText.js

**Pull request merged**